File: 917-boats-to-save-people/boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& wt, int limit) {
        int minNumBoats=0;
        sort(wt.begin(), wt.end());
        int beg=0, end=(int)wt.size()-1;
        while(beg<end){
            if(wt[beg]+wt[end]<=limit){
                ++beg;
            }
            ++minNumBoats;
            --end;
        }
        if(beg==end) ++minNumBoats;
        return minNumBoats;
    }
};

File: 917-boats-to-save-people/boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int lo=0, hi=people.length-1, boats=0;
        while(lo<hi){
            if(people[lo]+people[hi]<=limit){
                lo++; 
            }
            ++boats; hi--;
        }
        if(lo==hi) ++boats;
        return boats; 
    }
}